.dotnet/tests/Batch/BatchTests.cs

```diff
@@ -22,32 +23,71 @@ public BatchTests(bool isAsync) : base(isAsync)
 
-        long unixTime2024 = (new DateTimeOffset(2024, 01, 01, 0, 0, 0, TimeSpan.Zero)).ToUnixTimeSeconds();
+            long unixTime2024 = (new DateTimeOffset(2024, 01, 01, 0, 0, 0, TimeSpan.Zero)).ToUnixTimeSeconds();
+
+            foreach (JsonElement batchElement in dataElement.EnumerateArray())
+            {
+                JsonElement createdAtElement = batchElement.GetProperty("created_at");
+                long createdAt = createdAtElement.GetInt64();
+
+                Assert.That(createdAt, Is.GreaterThan(unixTime2024));
+            }
+            count++;
+
+            //var dynamicResult = result.GetRawResponse().Content.ToDynamicFromJson();
+            //Assert.That(dynamicResult.data.Count, Is.GreaterThan(0));
+            //Assert.That(dynamicResult.data[0].createdAt, Is.GreaterThan(new DateTimeOffset(2024, 01, 01, 0, 0, 0, TimeSpan.Zero)));
+        }
+
+        Assert.GreaterOrEqual(count, 1);
+    }
 
-        foreach (JsonElement batchElement in dataElement.EnumerateArray())
+    [Test]
+    public async Task ListBatchesProtocolAsync()
+    {
+        BatchClient client = GetTestClient();
+        IAsyncEnumerable<ClientResult> pageResults = client.GetBatchesAsync(after: null, limit: null, options: null);
+
+        int count = 0;
+        await foreach (ClientResult pageResult in pageResults)
         {
-            JsonElement createdAtElement = batchElement.GetProperty("created_at");
-            long createdAt = createdAtElement.GetInt64();
+            BinaryData response = pageResult.GetRawResponse().Content;
+            JsonDocument jsonDocument = JsonDocument.Parse(response);
+            JsonElement dataElement = jsonDocument.RootElement.GetProperty("data");
+
+            Assert.That(dataElement.GetArrayLength(), Is.GreaterThan(0));
+
+            long unixTime2024 = (new DateTimeOffset(2024, 01, 01, 0, 0, 0, TimeSpan.Zero)).ToUnixTimeSeconds();
+
+            foreach (JsonElement batchElement in dataElement.EnumerateArray())
+            {
+                JsonElement createdAtElement = batchElement.GetProperty("created_at");
+                long createdAt = createdAtElement.GetInt64();
+
+                Assert.That(createdAt, Is.GreaterThan(unixTime2024));
+            }
+            count++;
 
-            Assert.That(createdAt, Is.GreaterThan(unixTime2024));
+            //var dynamicResult = result.GetRawResponse().Content.ToDynamicFromJson();
+            //Assert.That(dynamicResult.data.Count, Is.GreaterThan(0));
+            //Assert.That(dynamicResult.data[0].createdAt, Is.GreaterThan(new DateTimeOffset(2024, 01, 01, 0, 0, 0, TimeSpan.Zero)));
         }
 
-        //var dynamicResult = result.GetRawResponse().Content.ToDynamicFromJson();
-        //Assert.That(dynamicResult.data.Count, Is.GreaterThan(0));
-        //Assert.That(dynamicResult.data[0].createdAt, Is.GreaterThan(new DateTimeOffset(2024, 01, 01, 0, 0, 0, TimeSpan.Zero)));
+        Assert.GreaterOrEqual(count, 1);
```

Comment:
I don't have a good feel for the semantics of this method, but if you know how many batches you're expecting you could validate that you get the right number of pages and elements in each page.  No worries if that's too complicated, though.

Reply:
@joseharriaga - Do you know where the data is coming from in this test? I don't see a create batch method in the test, but in the list method, there are 5 pages(4 pages with 20 elements on each page and 1 page with 17 elements). I'm not sure where that data is coming from.

I asserted greater than 1 because I'm not sure if 5 is always the correct number, and I'm also uncertain where this data is coming from.

Reply:
It's the batches that we have created over time.

Reply:
>list method returns 5 pages (4 pages with 20 elements each, and 1 page with 17 elements).

@joseharriaga - Just to confirm, these numbers aren't fixed, so we shouldn't assert specific values, correct? If we create more batches, the numbers will change, but asserting that the count is greater than 1 should work, right?

Reply:
Yes, that's correct